packages/access-client/src/agent-use-cases.js

```diff
@@ -201,12 +252,31 @@ export async function authorizeAndWait(access, email, opts) {
       { can: 'upload/*' },
     ]
   )
-  const sessionDelegations = [...(await expectAuthorization())]
+  const sessionDelegations = [...(await expectAuthorization(access, opts))]
   if (!opts?.dontAddProofs) {
     await Promise.all(sessionDelegations.map(async (d) => access.addProof(d)))
   }
 }
 
+/**
+ * Request authorization of a session allowing this agent to issue UCANs
+ * signed by the passed email address.
+ *
+ * @param {AccessAgent} accessAgent
+ * @param {`${string}@${string}`} email
+ * @param {object} [opts]
+ * @param {AbortSignal} [opts.signal]
+ * @param {Iterable<{ can: Ucanto.Ability }>} [opts.capabilities]
+ * @param {boolean} [opts.addProofs]
+ * @param {AuthorizationWaiter} [opts.expectAuthorization] - function that will resolve once account has confirmed the authorization request
+ */
+export async function authorizeWaitAndClaim(accessAgent, email, opts) {
+  await authorizeAndWait(accessAgent, email, opts)
+  await claimAccess(accessAgent, accessAgent.issuer.did(), {
+    addProofs: opts?.addProofs ?? true,
```

Comment:
Without this function having an `AndAdd` at the end, I'd still be surprised for it to default to add.
At the very least, the default to false should be documented on the `opts.addProofs` `@param` jsdoc?

Reply:
Hm, my problem is that from an external point of view, imho, `AndAdd` doesn't add anything useful beyond `Claim` here - there's no reason to claim delegations in a function that returns `void` if you don't save them.

Put another way, `AndAdd` begs the question "add to what" and the answer is an implementation detail that clients don't know anything about. I think the current name is clearer for users of this library and doesn't add a potentially confusing implementation detail to the list of things they need to think about - `Claim` suggests to them that the delegations they're looking for will be available after calling this function. 

All that said having a test here is a good idea, will add!

Reply:
ok test added! https://github.com/web3-storage/w3protocol/pull/666/commits/9146216fe53445b1858108272401a0c19cb6b944